pkg/engine/params_k8s.go

```diff
@@ -28,6 +28,11 @@ func assignKubernetesParameters(properties *api.Properties, parametersMap params
 		k8sComponents := api.K8sComponentsByVersionMap[k8sVersion]
 		kubernetesConfig := orchestratorProfile.KubernetesConfig
 		kubernetesImageBase := kubernetesConfig.KubernetesImageBase
+		hyperKubeImageBase := kubernetesConfig.KubernetesImageBase
+
+		if properties.IsAzureStackCloud() {
```

Comment:
@jackfrancis , the hyperkube image and other dependencies image are hosted in different repository for Azure Stack.  This change will use the information in kubernetesSpecConfig for the dependencies images and use kubernetesImageBase under kubernetesConfig for hyperkube images.